Document Firecracker's defense in depth approach (#2805)
firecracker-microvm/firecracker: docs/prod-host-setup.md

# Production Host Setup Recommendations

Firecracker relies on KVM and on the processor virtualization features

> for assuring workload isolation. Security guarantees and defense in deepth can

Reviewer: ```suggestion
for workload isolation. Security guarantees and defense in depth can
```

> only be upheald, if the following list of recommendations are implemented in

Reviewer: ```suggestion
only be upheld if the following list of recommendations are implemented in
```

## Jailer Configuration

For assuring secure isolation in production deployments, Firecracker should

> only be started only using the `jailer` binary that's part of each Firecracker

Reviewer 2: ```suggestion
must be started using the `jailer` binary that's part of each Firecracker
```

Author: Thanks for catching this typo. Now is resolved in a new iteration.

## Jailer Configuration

For assuring secure isolation in production deployments, Firecracker should
only be started only using the `jailer` binary that's part of each Firecracker
release, or executed under process constraints equal or more restrictive than

> those in the included jailer. For more about Firecracker sandboxing please see

Reviewer: ```suggestion
those in the jailer. For more about Firecracker sandboxing please see
```
